main.cpp

```diff
@@ -19,13 +19,39 @@ using namespace clang::tooling;
 
 class CastCallBack : public MatchFinder::MatchCallback {
 public:
-    CastCallBack(Rewriter& rewriter) {
-        // Your code goes here
-    };
+    CastCallBack(Rewriter& rewriter) : myrewriter(rewriter) {}
 
     void run(const MatchFinder::MatchResult &Result) override {
-        // Your code goes here
+    
+       		if (const auto *expr = Result.Nodes.getNodeAs<CStyleCastExpr>("cast")) {
+			auto type = Lexer::getSourceText(
+			CharSourceRange::getTokenRange(
+			expr->getLParenLoc().getLocWithOffset(1),
+                                    expr->getRParenLoc().getLocWithOffset(-1)),
+                                *Result.SourceManager, LangOptions());
+
+     auto cast_str = ("static_cast<"+type+">").str();
+
+	if(!isa<ParenExpr>(expr->getSubExprAsWritten())) // if no parenteses we need to add them
+	{
+		cast_str=("static_cast<"+type+">(").str(); // add left bracket
+	
+ 		myrewriter.InsertText(
+                        Lexer::getLocForEndOfToken(

```

Comment:
Please describe the reason of using the`Lexer::getLocForEndOfToken()` function here.
What would be the difference if `rewriter.InsertText(styleCastExpr->getEndLoc().getLocOffset(1), ")");` was used here?
What would be the difference if  `rewriter.InsertTextAfterToken(styleCastExpr->getEndLoc(), ")");` was used here?

Reply:
If we use rewriter.InsertText(styleCastExpr->getEndLoc().getLocOffset(1), ")"); then ")" will be inserted immediately after expr + 1(offset) if the variable name is longer than one character we will get wrong result.

i used Lexer::getLocForEndOfToken() to get the position after last token in the line;

Now i see that i could use rewriter.InsertTextAfterToken(expr->getEndLoc(), ")"); instead cause it gives the same result.